330 quickstart features docs (#356)
conda/conda-pypi: docs/quickstart.md

# Quickstart

## Installation

`conda-pypi` is a `conda` plugin that is available in your `base`
environment in conda versions 26.5 and newer.

Update your conda installation to get `conda-pypi`:

```bash
conda update conda
```

You can also install the plugin directly into your `base` environment:

```bash
conda install --name base conda-pypi
```

Once installed, the `conda pypi` subcommand becomes available across all your
conda environments.

## Set up the community wheel channel

> The conda-pypi community wheel channel is a public channel on anaconda.org

Reviewer: The channel is at https://repo.anaconda.cloud/conda-pypi instead of anaconda.org.

Reviewer 2: I left one above as well
```suggestion
The conda-pypi community wheel channel is a public channel hosted by Anaconda
```

that makes pure Python packages from PyPI available through `conda install`.
When you add this channel, conda's solver can find and install these packages
alongside your regular conda packages in a single step.

To enable the conda-pypi channel, configure the Rattler solver, add the channel, and
reset channel priority to its default (flexible):

> ```bash

Reviewer: To mirror these config steps, I think we should add a "Revert config" section in the quickstart to undo them as well.

Author: Good call

conda config --set solver rattler
conda config --append channels conda-pypi
conda config --set channel_priority flexible
```

With this configuration, `conda install` can resolve dependencies across both
regular conda packages and wheel packages in a single solve. When a wheel
package is selected, conda downloads the artifact directly from PyPI and
installs it into the environment while tracking it like any other conda
package.

> :::{admonition} Beta

Author: I should probably change this to experimental to match the other page? But it is technically a beta.

Reviewer: I think changing from experimental to beta is good 👍

:class: warning
The community wheel channel is in public beta. It hosts pure Python wheels
only. Compiled wheels are not supported. The security posture is the same as

> installing from public PyPI. For more details, see {ref}`community-wheel-channel`.

Reviewer 2: ```suggestion
The conda-pypi channel is in public beta. It hosts metadata only, for pure Python wheels from PyPI. Compiled wheels are not supported at the moment. 
The security posture is the same as installing from public PyPI. For more 
details, see {ref}`community-wheel-channel`.
```

:::

## Basic usage

`conda-pypi` provides several {doc}`features`. The main functionality is
accessed through the `conda pypi` command:

### Installing PyPI packages

:::{note}
These instructions assume that you have done the following:

- Created and activated a conda environment

> - Installed `python` and `pip` into that conda environment
> - Added the `conda-pypi` channel to your `.condarc` file
> - Configured your solver to be the rattler solver

Reviewer: ```suggestion
- Created and activated a conda environment
- Added the `conda-pypi` channel to your `.condarc` file
- Configured your solver to be the rattler solver
```
Adding python and pip in to your environment is only needed for the `conda pypi install` and `conda pypi convert` commands.

:::

> Use `conda install` to install a package (for example, `niquests`):

Reviewer: For `conda install` we should use an example package that isn't on conda-forge (only on PyPI) like django-modern-rest.

```bash
conda install niquests
```

This will download and unpack `niquests` from PyPI and
install it as a native wheel (`.whl`) file.
The dependencies of `niquests` will be installed from
the conda channel when available. For example, if `niquests` depends on
`urllib3` and `certifi`, and both are available on the conda channel, those
dependencies will be installed from conda rather than PyPI.

```bash

> conda install build

Reviewer: This section should be split up I think in to:
1. The primary `conda install <package>` workflow from the wheel channel
2. The secondary `conda pypi install` command. All of the text about packages coming from PyPI and filling in conda packages is for this command only.

For `conda install`, it works like any other channel. So channel priority and availability of packages determines what gets installed from where.
